widgets/nuxeo-directory-widget-behavior.html

```diff
@@ -0,0 +1,147 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<link rel="import" href="../../polymer/polymer.html">
+<link rel="import" href="../nuxeo-i18n-behavior.html">
+<link rel="import" href="../../iron-form-element-behavior/iron-form-element-behavior.html">
+<link rel="import" href="../../iron-validatable-behavior/iron-validatable-behavior.html">
+
+<script>
+  var Nuxeo = Nuxeo || {};
+  (function () {
+
+    /**
+     * `Nuxeo.DirectoryWidgetBehavior` provides helpers to build a widget to select Nuxeo vocabulary entrie(s).
+     *
+     * @polymerBehavior Nuxeo.DirectoryWidgetBehavior
+     */
+    Nuxeo.DirectoryWidgetBehavior = [Nuxeo.I18nBehavior, Polymer.IronFormElementBehavior, Polymer.IronValidatableBehavior, {
+      properties: {
+        /**
+         * Name of the directory.
+         */
+        directoryName: {
+          type: String
+        },
+
+        /**
+         * Checking this option means that the labels are localized with translations provided
+         * in the directory itself (i.e. in fields). Otherwise labels are translated as usual
+         * picking values in messages*.properties files.
+         */
+        dbl10n: {type: Boolean, value: false},
+
+        /**
+         * Label.
+         */
+        label: String,
+
+        /**
+         * In case of hierarchical vocabulary, if true, parent item can be selected.
+         */
+        canSelectParent: Boolean,
+
+        /**
+         * Set to `true` for read only mode.
+         */
+        readonly: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * Function used to get the id from the choice object.
+         */
+        idFunction: {
+          type: Function,
+          value: function() {
+            return this._idFunction.bind(this);
+          }
+        },
+
+        /**
+         * Error message to show when `invalid` is true.
+         */
+        errorMessage: String,
+
+        /**
+         * Formatter for the entries.
+         */
+        format: {
+          type: Function,
+          value: function() {
+            return this._formatter.bind(this);
+          }
+        },
+
+        /**
+         * Fired when the directory entries are loaded.
+         *
+         * @event loaded
+         */
+
+        _entries: Array
+      },
+
+      observers: [
+        '_fetchEntries(directoryName)'
+      ],
+
+      _fetchEntries: function() {
+        if (this.directoryName) {
+          this.value = [];
+          this.async(function() {
+            this.$.op.params = {
```

Comment:
We could probably use `document.createElement("nuxeo-operation")` on attached instead of expecting there to be a `<nuxeo-operation id="op">`

Reply:
TBH I really don't like this pattern as it hurts the element's code readability